libs/designsystem/src/lib/components/chart/chart.component.spec.ts

```diff
@@ -1,190 +1,47 @@
-        y: 20,
-        label: '20%',
-      },
-      {
-        name: 'Bubbles 41%',
-        y: 41,
-        label: '41%',
-      },
-    ];
-    component.ngOnInit();
-    const data = (component.options.series[0] as Highcharts.SeriesAreasplineOptions).data;
-    expect(data.length).toBe(2);
-    expect(data[0]['name']).toBe('Boomerangs 20%');
-  });
-
-  describe('ActivityGauge', () => {
-    it('should set correct title and subtitle', () => {
-      component.type = ChartType.ACTIVITYGAUGE;
-
-      component.data = [
-        {
-          title: '1.234.567',
-          subtitle: 'Afdraget',
-        },
-      ];
-
-      component.ngOnInit();
-
-      expect(component.options.title.text).toBe('1.234.567');
-      expect(component.options.subtitle.text).toBe('Afdraget');
-    });
-
-    it('should add backgroundColor to optionsarray', () => {
-      component.type = ChartType.ACTIVITYGAUGE;
-      const ActivityGaugeOptions: Options = {
-        pane: {
-          background: [
-            {
-              backgroundColor: {
-                linearGradient: { x1: 0, x2: 0, y1: 0, y2: 1 },
-                stops: [
-                  [1, 'rgba(255, 255, 255, 0.3)'],
-                  [0, 'rgba(255, 255, 255, 0.3)'],
-                ],
-              },
-              outerRadius: '112%',
-              innerRadius: '88%',
-              borderWidth: 0,
-            },
-          ],
-        },
-      };
-
-      component.options = ActivityGaugeOptions;
-
-      component.data = [
-        {
-          paneBackgroundColor: 'red',
-        },
-      ];
-
-      component.ngOnInit();
-
-      expect(component.options.pane.background[0].backgroundColor).toEqual(
-        component.data[0].paneBackgroundColor
-      );
-    });
+  beforeEach(() => {
+    chartHelper = new ChartHelper();
+    spyOn(chartHelper, 'updateChart');
 
-    it('should change title and subtitle color when color-attribute is set', () => {
-      component.type = ChartType.ACTIVITYGAUGE;
-      component.data = [
+    spectator = createHost('<kirby-chart [height]="height" [options]="options"></kirby-chart>', {
+      hostProps: { options: {} },
+      providers: [
         {
-          color: 'red',
+          provide: ChartHelper,
+          useValue: chartHelper,
         },
-      ];
-      const expected = component.data[0].color;
-
-      component.ngOnInit();
-
-      expect(component.options.title.style.color).toEqual(expected);
-      expect(component.options.subtitle.style.color).toEqual(expected);
+      ],
     });
+  });
 
-    it('should set type to solidgauge when ACTIVITYGAUGE is chosen', () => {
-      component.type = ChartType.ACTIVITYGAUGE;
-      component.data = [
-        {
-          title: '',
-          subtitle: '',
-        },
-      ];
+  it('should set correct height', () => {
+    spectator.setHostInput({ height: 200 });
 
-      component.ngOnInit();
+    const chartElm = spectator.query('[data-test=chart]'); // TODO: get height
```

Comment:
Please don't pollute the chart component with test attributes:
```suggestion
    const chartElm = spectator.element.firstChild;
```
OR
```suggestion
    const chartElm = spectator.query(':scope > div'); // TODO: get height
```

Reply:
Why not? We are needing to test against a GUI and this is a common best practice in gui testing seen in eg. Cypress best practices: https://docs.cypress.io/guides/references/best-practices.html.
Now obviously we are not using Cypress here, but I can't see why the same principle shouldn't be beneficial here?

Reply:
First child and :scope > div are very fragile selectors.